server/test.js

```diff
@@ -1,18 +1,44 @@
-import { AsyncLocalStorage } from 'node:async_hooks';
+import { AsyncLocalStorage } from "node:async_hooks";
+import { Worker } from "@temporalio/worker";
+import { Context } from "@temporalio/activity";
+import { Client } from "@temporalio/client";
+import { randomUUID } from "node:crypto";
 
-export function hereItStarts() {
-    const asyncLocalStorage = new AsyncLocalStorage();
+console.log("LOADING test.js");
 
-    function logWithId(msg) {
-      const id = asyncLocalStorage.getStore();
-      console.log(`${id !== undefined ? id : '-'}:`, msg);
-    }
+export async function hereItStarts() {
+  const asyncLocalStorage = new AsyncLocalStorage();
 
-    asyncLocalStorage.run(1, () => {
-      logWithId('start');
-      // Imagine any chain of async operations here
-      setImmediate(() => {
-        logWithId('finish');
-      });
+  function logWithId(msg) {
+    const id = asyncLocalStorage.getStore();
+    console.log(`${id !== undefined ? id : "-"}:`, msg);
+  }
+
+  asyncLocalStorage.run(1, () => {
+    logWithId("start");
+    // Imagine any chain of async operations here
+    setImmediate(() => {
+      logWithId("finish");
     });
+  });
+
+  const workflowId = "test" + randomUUID();
+  const client = new Client({});
+  await client.workflow.start("httpWorkflow", {
+    workflowId: workflowId,
+    taskQueue: "test",
+  });
+
+  const worker = await Worker.create({
+    taskQueue: "test",
+    activities: {
+      async testActivity() {
+        console.log(
+          `testActivity ${Context.current().info.workflowExecution.workflowId}`
+        );
+      },
+    },
+    maxConcurrentActivityTaskExecutions: 1,
```

Comment:
This line is very important. The mocked `AsyncLocalStorage` can only retain the context for one call at a time, so you need to make sure that the worker don't attempt to execute multiple activities concurrently.

If you really need to execute multiple activities concurrently from the Meteor process, then you may consider exploiting [asynchronous activity completion](https://docs.temporal.io/activities#asynchronous-activity-completion) to delegate actual execution of the activity to some meteor based function.